app/controllers/close_contacts_controller.rb

```diff
@@ -57,19 +54,44 @@ def update
 
   # Delete an existing close contact record
   def destroy
-    redirect_to root_url && return unless current_user.can_create_patient_close_contacts?
-    cc = current_user.get_patient(params.permit(:patient_id)[:patient_id])&.close_contacts&.find_by(id: params.permit(:id)[:id])
-    redirect_to root_url && return if cc.nil?
+    cc = current_user.get_patient(params.require(:patient_id)[:patient_id])&.close_contacts&.find_by(id: params.permit(:id)[:id])
     cc.destroy
     if cc.destroyed?
       reason = params.permit(:delete_reason)[:delete_reason]
       History.close_contact_edit(patient: params.permit(:patient_id)[:patient_id],
-                              created_by: current_user.email,
-                              comment: "User deleted a close contact (ID: #{cc.id}, Name: #{cc.first_name} #{cc.last_name}, Primary Telephone: "\
+                                 created_by: current_user.email,
+                                 comment: "User deleted a close contact (ID: #{cc.id}, Name: #{cc.first_name} #{cc.last_name}, Primary Telephone: "\
                               "#{cc.primary_telephone}, Email: #{cc.email}, Last Date of Exposure: #{cc.last_date_of_exposure}, "\
                               "Assigned User: #{cc.assigned_user}, Notes: #{cc.notes}, Contact Attempts: #{cc.contact_attempts}. Reason: #{reason}.")
     else
       render status: 500
     end
   end
+
+  private
+
+  def check_can_create
+    return head :forbidden unless current_user.can_create_patient_close_contact?
+  end
+
+  def check_patient
+    @patient_id = params.permit(:patient_id)[:patient_id]&.to_i
+
+    # Check if Patient ID is valid
+    unless Patient.exists?(@patient_id)
+      error_message = "Close contact cannot be deleted for unknown monitoree with ID: #{@patient_id}"
+      render(json: { error: error_message }, status: :bad_request) && return
+    end
+
+    # Check if user has access to patient
+    return if current_user.get_patient(@patient_id)
+
+    error_message = "User does not have access to Patient with ID: #{@patient_id}"
+    render(json: { error: error_message }, status: :forbidden) && return
+  end
+
+  def check_close_contact
+    @close_contact = CloseContact.find_by(id: params.permit(:id)[:id])
```

Comment:
use this here: `@close_contact = @patient.close_contacts.find_by_id(id: params.permit(:id)[:id])`